gammapy/makers/safe.py

```diff
@@ -247,6 +244,24 @@ def make_mask_energy_bkg_peak(dataset):
         energy_min = energy_axis.pix_to_coord(idx)
         return geom.energy_mask(energy_min=energy_min)
 
+    def make_mask_bkg_invalid(self, dataset):
+        """Mask non-finite values and zeros values in background maps
+ 
+        Parameters
+        ----------
+        dataset : `~gammapy.datasets.MapDataset`
+            Dataset to compute mask for.
+
+        Returns
+        -------
+        mask_safe : `~numpy.ndarray`
+            Safe data range mask.
+        """
+
+        bkg = dataset.background.data
+        mask = np.isfinite(bkg) & (bkg > 0.0)
```

Comment:
I just noticed there are test fails now for on-off datasets: it seems there are case where `alpha * n_off` is actually zero, because `n_off` is zero. @registerrier should those bins with `n_off` be included by default in stat computation? I remember there is a bias for this situation...

Reply:
I would say that we have  a poor background estimation if `n_off` is zero that would lead to over-estimate the overall signal-to-background. For now I have fixed the tests assuming that masking zeros is right.

Reply:
Sounds reasonable to me, maybe add an explanation in the docstring on the behaviour for on-off datasets. And maybe let's wait for @registerrier to finally comment...

Reply:
Technically, including the empty OFF bins can introduce a bias but removing them will introduce one as well.
Also if you use real ON-OFF 2D datasets you will end up having many empty bins.

I would not set such a mask by default on ON-OFF datasets.

Reply:
Thanks @registerrier, so it seems there is no obviously correct way to treat this. In this case we maybe just go with the "established" behaviour?  I guess traditionally people have included the bins and maybe we just do the same to not introduce unexpected behaviour for un-obvious reasons. Are you OK @QRemy to revert the change?

Reply:
I guess the issue here depends on what will be done later. Typically for an ON-OFF dataset, background estimation is carried out later. So this would not apply. 
I think we need to document that this is not to be applied for ON-OFF analysis.

Reply:
Yes, I think we have never defined (but maybe should...) on the order of background and safe data range estimation. I'm not even sure if there is a generally preferable order. Ad hoc I'd also say `SafeMaskMaker -> BackgroundMaker`...but there might be cases where the result of the background estimate affects the chosen data range...

Reply:
I will revert the changes and apply the bkg > 0 condition only if the dataset is not a ON-OFF.

Reply:
Thanks! @registerrier do we already have a note in the docs on the `wstat` bias? I guess we would need a reference which doesn't exist yet?